src/test/java/org/jenkinsci/plugins/workflow/test/steps/FakeTest.java

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.plugins.workflow.test.steps;
+
+import java.util.logging.Logger;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.graph.FlowGraphWalker;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class FakeTest {
+
+    @ClassRule
+    public static BuildWatcher bw = new BuildWatcher();
+
+    @Rule
+    public RestartableJenkinsRule rj = new RestartableJenkinsRule();
+
+    private static final Logger LOGGER = Logger.getLogger(FakeTest.class.getName());
+
+    @Test
+    public void fakeStepExecutes() {
+        rj.then(j -> {
+            WorkflowJob p = j.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("fake 20", true));
+            var run = p.scheduleBuild2(0).waitForStart();
+            j.waitForMessage("fake", run);
+            LOGGER.info("Jenkins will restart now");
+        });
+
+        rj.thenWithHardShutdown( j -> {
```

Comment:
Use `RealJenkinsRule`. There is a `stopForcibly` method if you actually need it (not sure).

Reply:
After using RealJenkinsRule the `simulatedSleepStep` is not found inside jenkins..
Checking if similar happens when using `JenkinsSessionRule`

Reply:
test works after switching to `JenkinsSessionRule` (logs in the PR description)

Reply:
> After using `RealJenkinsRule` the `simulatedSleepStep` is not found inside jenkins..

Sounds like there is something wrong with how you were registering it. You would need to use `addSyntheticPlugin`. Anyway if `JenkinsSessionRule` works, that should be much simpler.

Reply:
noted about the `addSyntheticPlugin` - i couldn't figure it out 😄; thank you so much for the info, will use in future.